### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by feyza-droid** [resuldagdanov/scenario-based-rl#3]
- [x] Move imitation learning evaluation trigger (and all related parts) to main.sh
- [x] Add stop traffic sign to calculate reward
- [x] Target deepcopy
- [x] Save resnet once and then load it each time
- [x] Solve deterministic reproducility problem occurs on resnet when is_cpu=False
- [x] Fix the problem with starting from route id other than 0
- [x] Extract all failed routes (with start and end waypoints)
- [x] Make sure resnet forward produces same values (e.g. image_features_torch) and reproducibility achieved (deterministic algorithms True)
- [x] Considering stop sign while rewarding may cause agent to stuck
- [x] Save all seeds to database and get them from database
- [x] Add moving average reward plot to tensorboard
- [x] For DQN, try different epsilon, target value update, gradient clipping and different DQNNetwork architectures
- [x] Save target value update, and epsilon step to db
- [x] For RL part (DQN) add early stopping
- [x] Remove all unused models/codes/hyperparams
- [x] Make sure leaderboard evaluater and scenario runner don't have randomness
----------------------------------------------------
- [x] DQN all failure cases training
- [x] DAgger
- [x] Classifier
- [x] Paper writing
    - [x] Collect previous papers
    - [x] Make visualization (for failure cases)
    - [x] Experiment results, comparison, tables
- [x] Benchmarks
